src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -859,6 +861,12 @@ def aks_upgrade(cmd,
         mc = client.get(resource_group_name, name)
         return _remove_nulls([mc])[0]
 
+    if tier is not None:
+        instance.sku.tier = tier
+
+    if k8s_support_plan is not None:
+        instance.support_plan = k8s_support_plan
```

Comment:
Is it necessary to add validation about lts tier requires premium sku?

Reply:
@FumingZhang thanks for reviewing! Yes, we can add it here or rely on the validation on the server side. I just updated the code to raise CLIError for it on L872.

Reply:
New live test passed: https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=98938062&view=results

Reply:
I am having an issue on L872. shouldn't it be instance.sku.tier not instance.tier?

Reply:
@sdrinovsky thanks for bringing this to my attention, I started a PR to get it fixed https://github.com/Azure/azure-cli/pull/29949